app/models/ext_management_system.rb

```diff
@@ -71,9 +71,9 @@ def self.supported_types_and_descriptions_hash
   def hostname_uniqueness_valid?
     return unless hostname_required?
     return unless hostname.present? # Presence is checked elsewhere
+    # check uniqueness per provider type
 
-    existing_hostnames = Endpoint.where.not(:resource_id => id).pluck(:hostname).compact.map(&:downcase)
-
+    existing_hostnames = (self.class.all - [self]).map(&:hostname).compact.map(&:downcase)
     errors.add(:hostname, "has already been taken") if existing_hostnames.include?(hostname.downcase)
```

Comment:
I think `dialog_tab.rb` is correct. In general, if a message (a validation message in this case) will be propagated into the UI, it needs to be inside `_()`. In some cases, it needs to be in `N_()` and
the actual translation is done by `_()` somewhere in the UI (controller or view), but that's a different
discussion really.